resources/js/components/ui/Table/ColumnName.tsx

```diff
@@ -0,0 +1,32 @@
+import {ColumnDef} from "@tanstack/solid-table";
+import {Component, Show} from "solid-js";
+import {Capitalize} from "../Capitalize";
+import {useTable} from "./TableContext";
+
+interface Props {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  def: ColumnDef<any>;
+}
+
+/**
+ * Component displaying the column name, taken from column meta.columnName if present,
+ * otherwise from table meta.translations, or finally from the column id.
+ */
+export const ColumnName: Component<Props> = (props) => {
+  const table = useTable();
+  return (
+    <Show
+      when={props.def.meta?.columnName}
+      fallback={
+        <Show
+          when={table.options.meta?.translations?.headers?.(props.def.id || "", {defaultValue: ""})}
+          fallback={props.def.id}
+        >
+          {(columnName) => <Capitalize text={columnName()} />}
+        </Show>
+      }
+    >
+      {(columnName) => <>{columnName()()}</>}
```

Comment:
co daje <> i dlaczego ()() ? czy mogłoby być:

`
<Show when={props.def.meta?.columnName} fallback=... > {props.def.meta?.columnName} </Show>
`

albo

`
<Show when={props.def.meta?.columnName} fallback=... > {(columnName) => columnName} </Show>
`

?

Reply:
Funkcyjna wersja Show bierze jako children funkcję, ale nie przekazuje do niej wartości obliczonej z when, tylko accessor do tej wartości (`Accessor<T>=()=>T`). Tak jest z reaktywnościowych powodów lepiej, na przykład jak podam do tego Show funkcję `x => <span>{x()}</span>` to span jest utworzony tylko raz, a następnie jego wnętrze reaguje na zmiany wartości when. Gdyby Show przekazywał jako x wartość z when (a nie accessor do niej) to span by się tworzył przy każdej zmianie x, a to jest niepotrzebne.

A że już samo props.columnName jest typu `()=>JSX.Element` no to `()()`. A columnName jest takiego typu (a nie JSX.Element) dlatego że gdyby to był element, mógłby znajdować się tylko w jednym miejscu w DOMie, a tak to mogę chcieć mieć więcej jego kopii, no i tak się nawet dzieje. Oraz reaktywność.

Natomiast jak chodzi o `<>` to zwykle służy do grupowania elementów JSX, czyli `<span/><span/>` nie jest poprawnym JSX, musi być jeden top level element, więc piszę `<><span/><span/></>`. Tutaj używam tego do czego innego, i jest to związane z (nie zgadniesz) reaktywnością. Dajmy na to że columnName jako ten JSX.Element zwraca zwykłego stringa. Wtedy ta funkcja dostaje accessor do accessora do tego stringa i zwraca tego stringa, zwykłego lamerskiego stringa bez żadnej reaktywności, więc oczywiście jeśli wartość columnName się zmieni to on nie zareaguje. Natomiast jeśli zwracam `<>{coś()}</>` to już jest reaktywne bo to solid ewaluuje sobie tego stringa i rozumie że on ma zależności. Tu zobacz demo: https://playground.solidjs.com/anonymous/466264e5-0a85-48c0-928d-280ac245cf4d

Twój pierwszy kod by mógł być, plus <> i jeszcze () za columnName. Więc nie wiem czy warto, nadal dużo krzaków :)